### PR TITLE
Signup use case frontend. 

### PR DESCRIPTION
**Pull request by stefanateljevic04**
Implemented all of the required components of the view for the signup use case. Created a main method in the signupview class to test that the view works, shown in screenshot:

Screen Shot 2023-11-25 at 2 23 08 PM

@vrindasubhash @merrickliu888 @jackkfan0305 @samaslanowicz can you guys review this?

### CODE REVIEW

**Comment by merrickliu888** [comment]
@stefanateljevic04 Looks solid. But what is the purpose of clear? And could you change the name of the text boxes to what they would look like on say a gmail sign up ie. instead of "Choose username" just do Username. Also get rid of the Sign Up View header. But def looking good!

**Comment by stefanateljevic04** [comment]
@merrickliu888 I made the changes, here's the new screenshot:

Screen Shot 2023-11-26 at 2 25 28 AM

**Comment by stefanateljevic04** [comment]
@merrickliu888 should I get rid of cancel also?

**Comment by merrickliu888** [comment]
> @merrickliu888 should I get rid of cancel also?

Does it have any functionality. It would also be useful to have a button like "already have an account? Sign in"

**Comment by stefanateljevic04** [comment]
> > @merrickliu888 should I get rid of cancel also?
> 
> Does it have any functionality. It would also be useful to have a button like "already have an account? Sign in" Unless thats what cancel is for. We do not want users being stuck on the signup page.

That makes sense, I'l turn cancel into that

**Comment by stefanateljevic04** [comment]
> > @merrickliu888 should I get rid of cancel also?
> 
> Does it have any functionality. It would also be useful to have a button like "already have an account? Sign in" Unless thats what cancel is for. We do not want users being stuck on the signup page.

Ok I made the changes, here's what it looks like now 

Screen Shot 2023-11-27 at 2 01 05 AM

File: src/view/SignupView.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            public void keyPressed(KeyEvent e) {
+            }
+
+            public void keyReleased(KeyEvent e) {
+            }
+        });
+        this.repeatPasswordInputField.addKeyListener(new KeyListener() {
+            public void keyTyped(KeyEvent e) {
+                SignupState currentState = signupViewModel.getState();
+                String var10001 = SignupView.this.repeatPasswordInputField.getText();
```
Comment (merrickliu888) [inline src/view/SignupView.java:94]: What is `var10001`?

File: src/view/SignupView.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                SignupState currentState = signupViewModel.getState();
+                this.signupController.execute(currentState.getUsername(), currentState.getPassword(), currentState.getRepeatPassword());
+            }
+
+        });
+
+        this.usernameInputField.addKeyListener(new KeyListener() {
+            public void keyTyped(KeyEvent e) {
+                SignupState currentState = signupViewModel.getState();
+                String var10000 = SignupView.this.usernameInputField.getText();
```
Comment (merrickliu888) [inline src/view/SignupView.java:65]: What is `var10000`?

**Review by merrickliu888 — CHANGES_REQUESTED** [review]
Awesome! Just need to change some variable names!

**Review by merrickliu888 — APPROVED** [review]
Solid. Now we just need to do some integration tests.

**Pull request merged**